src/ucp/core/ucp_context.c

```diff
@@ -1565,12 +1565,13 @@ ucp_add_component_resources(ucp_context_h context, ucp_rsc_index_t cmpt_index,
             }
 
             ucs_memory_type_for_each(mem_type) {
-                if ((context->config.ext.reg_nb_mem_types & UCS_BIT(mem_type)) &&
-                    !(md_attr->reg_nonblock_mem_types & UCS_BIT(mem_type))) {
-                    continue;
-                }
-
                 if (md_attr->flags & UCT_MD_FLAG_REG) {
+                    if ((context->config.ext.reg_nb_mem_types & UCS_BIT(mem_type)) &&
+                        !(md_attr->reg_nonblock_mem_types & UCS_BIT(mem_type))) {
+                        context->reg_block_md_map[mem_type] |= UCS_BIT(md_index);
```

Comment:
seems we don't check md_attr->reg_mem_types here, which leads for example to registering host memory with gdr_copy